ntoskrnl/ke/bug.c

```diff
@@ -334,6 +334,420 @@ KiInitializeBugCheck(VOID)
     }
 }
 
+PCHAR
+NTAPI
+KeGetBugMessageName(IN ULONG MessageId)
```

Comment:
I am totally sure that this incorrect way to get bugcheck code name. This code is almost two years old and I lost that tool source code lol.

Reply:
You probably Want to use RtlFindMessage, like what is used here:
https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/ex/init.c;hb=89e716a7f71c57d5c46718feec6f9f562fc121d4#l1754

Reply:
`RtlFindMessage` would only work if the bugcheck message description does only have the actual bugcheck title itself, which not all of them have it unfortunately. Consider this e.g.:
https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/reactos/mc/bugcodes.mc;h=329b75a2e4e94631d5b102a1c13386e8d7256fa2;hb=HEAD#l1718

So you'd end up with `A critical kernel security check failed` instead of the bugcheck itself. Maybe adding an additional type block like `BugcheckTitle` in .mc and then implement a function like `RtlFindBugcheckTitle` that is similar to `RtlFindMessage` would help?

Reply:
> So you'd end up with `A critical kernel security check failed` instead of the bugcheck itself. Maybe adding an additional type block like `BugcheckTitle` in .mc and then implement a function like `RtlFindBugcheckTitle` that is similar to `RtlFindMessage` would help?

This does not make sense. The .mc files are compiled and generate binary message resources (see how these look like when you open e.g. ntoskrnl.exe in a resource viewer/editor) and corresponding header files containing the list of #defines for their values. These keywords (SymbolicName etc...) are interpreted by the message compiler (mc.exe in Visual C++, windmc.exe in mingw etc.), and at best, adding a new "block" would do nothing at all, at worst it would emit an error saying the "block" is not recognized and fail. In any case you would not gain any thing at all.